### DISCUSSION TITLE
Cache occupancy grid, and take in delta updates

### DISCUSSION THREAD

**Discussion by bhanug12** [foxglove/community#605] [Q&A]
Hi all, while working with OccupancyGrid, one is visualizing a robot moving to new territories. In this case, new (map-tiles, offset) are sent where map-tile is a small patch of new/updated map, and offset is the offset with respect to the coordinate system. It is kind of like cache system. The benefits are that not the entire map would have to be sent over websocket or stored in mcap every instant (which mostly is the same information). Only a small portion of the map is updated every instant. 

Currently to display a map, one has to send the entire map every instant which wastes network bandwidth and increases storage space. 

Thanks you!

**Comment by amacneil** [comment]
Are the new tiles loaded from the robot or recording via a typical ros/mcap topic, or are they loaded from an external http service?

What message schemas do you use to represent these map updates? Is there a ros package to handle that, or something custom you wrote?
